.github/workflows/cancel_workflows_for_pr.yaml

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2024 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Cancel workflow on PR
+on:
+    workflow_dispatch:
+      inputs:
+        pull_request_id:
+          description: 'PR number to consider'
+          required: true
+          type: number
+        commit_sha:
+          description: 'Cancel runs for this specific SHA'
+          required: true
+          type: string
+
+jobs:
+    cancel_workflow:
+        name: Report on pull requests
+
+        runs-on: ubuntu-latest
+
+        # Don't run on forked repos
+        if: github.repository_owner == 'project-chip'
+
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v4
+            - uses: actions/setup-python@v5
+              with:
+                python-version: '3.12'
+            - name: Setup pip modules we use
+              run: |
+                  pip install     \
+                      click       \
+                      coloredlogs \
+                      pygithub    \
+                      && echo "DONE installint python prerequisites"
+            - name: Cancel runs
+              run: |
+                  scripts/tools/cancel_workflows_for_pr.py       \
+                    --pull-request ${{ inputs.pull_request_id }} \
+                    --commit-sha "${{ inputs.commit_sha }}"      \
+                    --token "${{ secrets.GITHUB_TOKEN }}"
```

Comment:
I would consider using a different name, maybe more specific to that purpose, so noone will elevate permission for that token. The case is, that someone might create PR with some typo fix, which for sure will be accepted. Then such person will become collaborator, so CI will run without approval. Then such eve-collaborator might change `cancel_workflows_for_pr.py` to dump the secret to output (after obfuscation of course).... PR will fail and trigger cancellation... and repo might be compromised.

Reply:
> I am unsure if/how we can mitigate this as we need at least some "cancel workflows" access.

I've just checked how `secrets.GITHUB_TOKEN` and it seems to be short-lived token which is generated for the workflow only. I was not aware of that.
>The GITHUB_TOKEN expires when a job finishes or after a maximum of 24 hours.